### PR TITLE
Added TeaLeaf Package

### PR DESCRIPTION
**Pull request by homerdin**
New Package.  TeaLeaf Proxy Application

### CODE REVIEW

File: var/spack/repos/builtin/packages/tealeaf/package.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        if '%gcc' in spec:
+            self.build_targets.extend(['COMPILER=GNU'])
+        elif '%cce' in spec:
+            self.build_targets.extend(['COMPILER=CRAY'])
+        elif '%intel' in spec:
+            self.build_targets.extend(['COMPILER=INTEL'])
+        elif '%pgi' in spec:
+            self.build_targets.extend(['COMPILER=PGI'])
+        elif '%xl' in spec:
+            self.build_targets.extend(['COMPILER=XL'])
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/tealeaf/package.py]: If all you need to do is override `build_targets`, you can do it like so:
```python
@property
def build_targets(self):
    targets = [
        '--directory=TeaLeaf_ref',
        'MPI_COMPILER={0}'.format(spec['mpi'].mpifc),
        'C_MPI_COMPILER={0}'.format(spec['mpi'].mpicc),
    ]

    if '%gcc' in spec:
        targets.append('COMPILER=GNU')
    ...

    return targets
```

File: var/spack/repos/builtin/packages/tealeaf/package.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+class Tealeaf(MakefilePackage):
+    """Proxy Application. TeaLeaf is a mini-app that solves
+       the linear heat conduction equation on a spatially decomposed
+       regularly grid using a 5 point stencil with implicit solvers.
+    """
+
+    homepage = "http://uk-mac.github.io/TeaLeaf/"
+    url      = "http://mantevo.org/downloads/releaseTarballs/miniapps/TeaLeaf/TeaLeaf-1.0.tar.gz"
+
+    tags = ['proxy-app']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/tealeaf/package.py]: What is this used for?

File: var/spack/repos/builtin/packages/tealeaf/package.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        # Manual Installation
+        mkdirp(prefix.bin)
+        mkdirp(prefix.doc.tests)
+
+        install('README.md', prefix.doc)
+        install('TeaLeaf_ref/tea_leaf', prefix.bin)
+        install('TeaLeaf_ref/tea.in', prefix.bin)
+
+        for f in glob.glob('TeaLeaf_ref/*.in'):
+            install(f, prefix.doc.tests)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/tealeaf/package.py]: You can replace this for-loop with:
```python
install_tree('TeaLeaf_ref', prefix.doc.tests)
```
Then you won't need to import glob either.

Comment (homerdin) [inline var/spack/repos/builtin/packages/tealeaf/package.py]: Was using glob to install only the data input files that are in the the same directory as the source code.

File: var/spack/repos/builtin/packages/tealeaf/package.py (hunk 2 above)
Comment (homerdin) [inline var/spack/repos/builtin/packages/tealeaf/package.py]: This was suggested as a way to help find the proxy apps.  Todd said they would be able to add support for it.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/tealeaf/package.py]: I don't really like adding special variables until we actually have support for them, but I'll let Todd decide. In the meantime, you can get what you want with:
```
$ spack list --search-description 'proxy app'
```

**Pull request merged**